### ISSUE TITLE
HGCN training lacks dynamic curriculum learning

### ISSUE THREAD

**Issue by lowmason** [lowmason/naics-embedder#63]
## Summary

The text model training uses a multi-phase curriculum (e.g., weighting nearer negatives early, full hard mining later), but HGCN training runs for a fixed number of epochs with static settings.

## Current Behavior

- HGCN trains with constant hyperparameters throughout
- No staged introduction of harder negatives
- No progression in margin or mining strategy

## Suggested Improvement

Implement a lightweight curriculum for HGCN:
```python
# Example curriculum phases
if epoch < warmup_epochs:
    # Phase 1: only sample negatives within k hops
    negatives = sample_nearby_negatives(anchor, max_hops=3)
    margin = base_margin * 0.5
else:
    # Phase 2: full negative sampling with hard mining
    negatives = sample_hard_negatives(anchor, all_nodes)
    margin = base_margin
```

### Rationale

- Early training benefits from stable gradients (nearby negatives)
- Later training benefits from harder examples to refine boundaries
- Mirrors the successful curriculum in the text model

## Priority

Low—HGCN is small (few layers, pre-learned input embeddings), so this is a nice-to-have rather than critical.
